scripts/qesap/lib/config.py

```diff
@@ -20,58 +40,76 @@ def yaml_to_tfvars(yaml_data):
     """
     config_out = ''
     terraform_variables = yaml_data['terraform']['variables']
-    print(yaml_data)
+    log.debug(yaml_data)
     for key, value in terraform_variables.items():
-        if isinstance(value, (str, int)):
-            entry = f'{key} = "{str(value)}"'
-        elif isinstance(value, bool):
-            entry = f'{key} = "{str(value).lower()}"'
-        elif isinstance(value, list):
-            entry = '", "'.join(value)
-            entry = f'{key} = ["{entry}"]'
-        elif isinstance(value, dict):
-            param_value = ''
-            for dict_key, dict_value in value.items():
-                param_value = f'{param_value}\t{dict_key} = "{dict_value}"\n'
-            entry = f'{key} = {{\n' \
-                    f'{param_value}' \
-                    f'}}'
-        else:
+        res, entry = yaml_to_tfvars_entry(key, value)
+        if not res:
             log.error(f'Unrecognized value type in yaml file: {key} = {value}')
-            return False
-        config_out = f'{config_out}\n{entry}'
+            return False, ''
+        config_out += f'\n{entry}'
 
     return config_out
 
 
+def terraform_yml(configure_data):
+    """
+    Check if Terraform:variables are present in the config.yaml
+    """
+    if not configure_data:
+        log.error("No configure_data")
+        return False
+
+    if 'terraform' not in configure_data.keys():
+        log.error("Missing 'terraform' key in configure_data")
+        return False
+    
+    if configure_data['terraform'] is None:
+        log.error("configure_data['terraform'] is empty")
+        return False
+
+    if 'variables' not in configure_data['terraform'].keys():
+        log.error("Missing 'variables' key in configure_data['terraform'] ")
+        return False
+
+    return True
+
+
 def template_to_tfvars(tfvars_template, configure_data):
-    """ takes
+    """ 
+    Takes data structure collected from yaml config. 
+    Values are converted into tfvars format and checked against terraform.tfvars.template.
+    Variables from yaml config are overwritten by values from template file.
 
     Args:
-        configure_data (dict): configuration data sctructure
+        configure_data (dict): configuration data structure
         tfvars_template (str): path to the tfvars template file
 
     Returns:
         bool: True(pass)/False(failure)
     """
+    log.info("Read %s", tfvars_template)
     with open(tfvars_template, 'r') as f:
-        tfvar_content = f.readlines()
+        tfvar_content = ['{}\n'.format(line.rstrip()) for line in f.readlines()]
```

Comment:
To pacify latest Pylint:

```suggestion
        tfvar_content = [f"{line.rstrip()}\n" for line in f.readlines()]
```